### PR TITLE
Make use of updated `trip-details` component

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Blocked by https://github.com/opentripplanner/otp-ui/pull/294. **Tests will fail until new packages are released.**

This PR uses the updated `trip-details` component (https://github.com/opentripplanner/otp-ui/pull/294) to display all fares returned by OTP.
Screen Shot 2021-10-05 at 4 23 03 PM

### CODE REVIEW

File: lib/components/narrative/default/default-itinerary.js [hunk 1]
```diff
@@ -23,6 +28,11 @@ const { isBicycle, isMicromobility, isTransit } = coreUtils.itinerary
 
 // Styled components
 const LegIconWrapper = styled.div`
+  display: inline-block;
```
Comment (philip-cline) [inline lib/components/narrative/default/default-itinerary.js:31]: Has this CSS be duplicated?

File: lib/components/narrative/line-itin/itin-summary.js [hunk 2]
```diff
@@ -86,14 +86,15 @@ export class ItinerarySummary extends Component {
...
     const { LegIcon } = this.context
 
     const {
       maxTNCFare,
       minTNCFare,
-      transitFare
-    } = coreUtils.itinerary.calculateFares(itinerary)
+      transitFares
+    } = coreUtils.itinerary.calculateFares(itinerary, true)
+    const transitFare = (transitFares?.[defaultFare] ?? transitFares.regular).transitFare
```
Comment (philip-cline) [inline lib/components/narrative/line-itin/itin-summary.js:97]: Is optional chaining here necessary since `transitFares.regular` is referenced directly after?

**Review by philip-cline — APPROVED** [review]
Just a few small comments

File: lib/components/narrative/default/default-itinerary.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/default-itinerary.js:31]: This was added by merging dev into this branch, so I'm not quite sure why it's appearing as a change!

File: lib/components/narrative/line-itin/itin-summary.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/line-itin/itin-summary.js:97]: Good catch! Got myself confused here while bug fixing. Have now replaced this with a more standard `||`. It's for situations where `defaultFare` isn't `regular`

File: lib/components/narrative/default/default-itinerary.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/default-itinerary.js]: Addressed in 88c0af89a48df3baed5064b7a4dd86720c3ca06d

File: lib/components/narrative/line-itin/itin-summary.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/line-itin/itin-summary.js]: Addressed in 88c0af89a48df3baed5064b7a4dd86720c3ca06d

**Review by philip-cline — APPROVED** [review]

File: lib/components/narrative/connected-trip-details.js [hunk 3]
```diff
@@ -13,6 +13,8 @@ const TripDetails = styled(TripDetailsBase)`
 
 const mapStateToProps = (state, ownProps) => {
   return {
+    defaultFare: state.otp.config.itinerary?.defaultFare || undefined,
+    fareKeyNameMap: state.otp.config.itinerary?.fareKeyMap || {},
     longDateFormat: coreUtils.time.getLongDateFormat(state.otp.config),
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/connected-trip-details.js:18]: Add examples in `example-config.yml`

File: lib/components/narrative/connected-trip-details.js [hunk 4]
```diff
@@ -13,6 +13,8 @@ const TripDetails = styled(TripDetailsBase)`
 
 const mapStateToProps = (state, ownProps) => {
   return {
+    defaultFare: state.otp.config.itinerary?.defaultFare || undefined,
+    fareKeyNameMap: state.otp.config.itinerary?.fareKeyMap || {},
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/connected-trip-details.js:17]: Also rename `state.otp.config.itinerary?.fareKeyMap` -> `.fareKeyNameMap`

File: lib/components/narrative/connected-trip-details.js [hunk 5]
```diff
@@ -13,6 +13,8 @@ const TripDetails = styled(TripDetailsBase)`
 
 const mapStateToProps = (state, ownProps) => {
   return {
+    defaultFare: state.otp.config.itinerary?.defaultFare || undefined,
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/connected-trip-details.js:16]: Rename to `defaultFareKey` per https://github.com/opentripplanner/otp-ui/blob/master/packages/trip-details/src/index.js#L204

File: lib/components/narrative/default/default-itinerary.js [hunk 6]
```diff
@@ -27,37 +32,42 @@ const LegIconWrapper = styled.div`
   height: 20px;
   padding-bottom: 6px;
   padding-left: 2px;
-  width: 20px;
+  width: 20px;
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/default/default-itinerary.js:35]: Why was the trailing space added?

File: lib/components/narrative/default/default-itinerary.js [hunk 7]
```diff
@@ -141,39 +158,46 @@ const ITINERARY_ATTRIBUTES = [
...
+    const visibleNotSet =
+      visibleItinerary === null || visibleItinerary === undefined
+    const isVisible =
+      visibleItinerary === index || (active === index && visibleNotSet)
     if (typeof setVisibleItinerary === 'function' && !isVisible) {
-      setVisibleItinerary({index})
+      setVisibleItinerary({ index })
     }
-  }
+  };
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/default/default-itinerary.js:170]: Remove extra semicolons in this file.

File: lib/components/narrative/default/default-itinerary.js [hunk 8]
```diff
@@ -209,8 +233,7 @@ class DefaultItinerary extends NarrativeItinerary {
               const bSelected = this._isSortingOnAttribute(b)
               if (aSelected) return -1
               if (bSelected) return 1
-
-              return a.order - b.order
+              else return a.order - b.order
```
Comment (binh-dam-ibigroup) [inline lib/components/narrative/default/default-itinerary.js:236]: Revert this change.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please update version numbers, make sure the names of passed props to TripDetails are correct. Also address instance of strange formatting and fix tests.

File: lib/components/narrative/connected-trip-details.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/connected-trip-details.js]: Updated in a32088189205a527c35064378e6ef614edd5b98e

File: lib/components/narrative/connected-trip-details.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/connected-trip-details.js]: Updated in a32088189205a527c35064378e6ef614edd5b98e

File: lib/components/narrative/connected-trip-details.js (hunk 5 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/connected-trip-details.js]: Updated in a32088189205a527c35064378e6ef614edd5b98e

File: lib/components/narrative/default/default-itinerary.js (hunk 6 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/default-itinerary.js]: I ran prettier at some point which caused these changes. Removed in 513973b0fc2d4404796775573af65c8859c746a2

File: lib/components/narrative/default/default-itinerary.js (hunk 7 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/default-itinerary.js]: I ran prettier at some point which caused these changes. Removed in 513973b

File: lib/components/narrative/default/default-itinerary.js (hunk 8 above)
Comment (miles-grant-ibigroup) [reply lib/components/narrative/default/default-itinerary.js]: I ran prettier at some point which caused these changes. Removed in 513973b

**Comment by miles-grant-ibigroup** [comment]
Thanks Binh! All good catches, all are addressed, with all relevant otp-ui packages updated. Have tested locally and fares are appearing correctly!

**Review by binh-dam-ibigroup — APPROVED** [review]
Good job!
